.pipelines/e2e.yml

```diff
@@ -18,190 +20,299 @@ pr:
-
-      - script: |
-          . secrets/env
-
-          # Retrieve the kubeconfig
-          hack/get-admin-aks-kubeconfig.sh > aks.kubeconfig
-
-          if [ -f aks.kubeconfig ]; then
-            echo "Kubeconfig retrieved successfully."
-          else
-            echo "Failed to retrieve Kubeconfig."
-            exit 1
-          fi
-        displayName: Get Kubeconfig for AKS Cluster
-
-      - script: |
-          az account set -s $AZURE_SUBSCRIPTION_ID
-          SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME) make secrets
-          . secrets/env
-
-          export CI=true
-          . ./hack/e2e/run-rp-and-e2e.sh
-          deploy_e2e_db
-
-          export SKIP_MIWI_ROLE_ASSIGNMENT=true
-          . ./hack/devtools/local_dev_env.sh
-          create_miwi_env_file
-        displayName: Setup (Azure)
-
-      - script: |
-          export CI=true
-          export ARO_E2E_MIMO=true
-          export USE_WI=true
-          export ARO_INSTALL_VIA_HIVE="true"
-          export ARO_ADOPT_BY_HIVE="true"
-          export HIVE_KUBE_CONFIG_PATH="$(realpath ./aks.kubeconfig)"
-
-          . ./env
-          . secrets/env
-          . ./hack/e2e/run-rp-and-e2e.sh
-
-          az acr login --name arosvcdev
-          az acr login --name arointsvc
-
-          run_vpn
-
-          run_podman
-          validate_podman_running
-
-          run_portal
-          validate_portal_running
-
-          run_selenium
-          validate_selenium_running
-
-          update_role_sets
-
-          run_mimo_actuator
-          validate_mimo_actuator_running
-
-          run_rp
-          validate_rp_running
-
-          register_sub
-
-          # cluster is cleaned up in a later step. Keep it around for now to enable must-gather collection on error
-          export E2E_DELETE_CLUSTER=false
-          make test-e2e -o e2e.test
-        displayName: Execute Tests
-        target: container
-
-      - script: |
-          export CI=true
-          . ./hack/e2e/run-rp-and-e2e.sh
-          set -x
-
-          # retrieve the kubeconfig
-          hack/get-admin-kubeconfig.sh /subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$CLUSTER/providers/Microsoft.RedHatOpenShift/openShiftClusters/$CLUSTER >admin.kubeconfig
-        displayName: Get admin kubeconfig for must-gather
-        condition: failed()
-      # must-gather collection must be run inside the container so it can access the VPN
-      - script: |
-          export CI=true
-          . ./hack/e2e/run-rp-and-e2e.sh
-
-          export KUBECONFIG=admin.kubeconfig
-
-          # retrieve the oc cli
-          wget -nv https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/$(OpenShiftCLIVersion)/openshift-client-linux-$(OpenShiftCLIVersion).tar.gz
-          tar xf openshift-client-linux-$(OpenShiftCLIVersion).tar.gz
+      TAG: $[ stageDependencies.Set_Tag_Stage.Set_Tag_Variable.outputs['SetTag.TAG'] ]
+    jobs:
+      - job: Run_E2E_Tests
```

Comment:
My recommended approach to start would be to just copy/paste the existing CSP e2e verbatim: https://github.com/Azure/ARO-RP/blob/master/.pipelines/ci.yml#L174-L292

Then add the steps required for MIWI to the step where we run the docker compose (but before we run it):

- export USE_WI=true
- export SKIP_MIWI_ROLE_ASSIGNMENT=true
- create_miwi_env_file
- update_role_sets
- `. ./env`
- any others I'm forgetting :)